src/Microsoft.SqlTools.SqlCore/ObjectExplorer/StatelessObjectExplorer.cs

```diff
@@ -37,78 +38,49 @@ public static async Task<TreeNode[]> Expand(string connectionString, SecurityTok
         {
             using (SqlConnection conn = new SqlConnection(connectionString))
             {
-                ServerConnection connection;
-                if (accessToken != null)
-                {
-                    connection = new ServerConnection(conn, accessToken as IRenewableToken);
-                }
-                else
-                {
-                    connection = new ServerConnection(conn);
-                }
-
-                try
-                {
-                    return await Expand(connection, accessToken, nodePath, serverInfo, options, filters);
-                }
-                finally
-                {
-                    if (connection.IsOpen)
-                    {
-                        connection.Disconnect();
-                    }
-                }
+                conn.AccessToken = accessToken?.Token;
+                conn.Open();
+                ServerConnection connection = new ServerConnection(conn);
+                connection.Connect();
```

Comment:
>   connection.Connect(); [](http://example.com/codeflow?start=13&length=24)

There are 2 changes here, really. 
1. You are setting the token explicitly on the SqlConnection instead of using IRenewableToken
2. You call SqlConnection.Connect explicitly instead of relying on ServerConnection to call it.

What happens if you just make change 1? Did this bug manifest only in AAD scenarios and not in SQL auth?
The behavior of ServerConnection re: the connection state is supposed to be auth-type-independent.

90% of SMO application code never calls `SqlConnection.Open` explicitly and relies on ServerConnection to call it as needed.

I suspect either your IRenewableToken implementation has a bug, or ServerConnection has a regression in the IRenewableToken code path. SSMS stopped using IRenewableToken a while back so the latter scenario is certainly a possibility.

Reply:
Also - have you considered implementing a custom interactive authentication provider to provide tokens directly to SqlClient instead of relying on IRenewableToken?

Reply:
whatever the issue is in ServerConnection - please go to the internal ADO repo for it and write a test for the expected behavior and see if it's a regression that should be fixed.

Reply:
Have you tried enabling 'EnableSqlAuthenticationProvider' on target database?

Reply:
@cheenamalhotra , I am just using sql core package for fabric which does not contain Connection service yet. Therefore I can not use 'EnableSqlAuthenticationProvider'

Reply:
You could still register auth provider on SqlConnection (required only once globally) and consume it from Authentication project - but I'll leave it up-to you to decide.